### PR TITLE
Implement multinomial

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented `cupy.random.multinomial`. Please merge #84 first.

### CODE REVIEW

File: cupy/random/sample.py [hunk 1]
```diff
@@ -164,3 +168,45 @@ def choice(a, size=None, replace=True, p=None):
...
+
+    p = len(pvals)
+    xs = choice(p, p=pvals, size=n * m)
+    shape = size + (p,)
+    ys = basic.zeros(shape, 'i')
+    core.ElementwiseKernel(
+        'int64 x, int32 p, int32 n', 'raw int32 ys',
+        'atomicAdd(&ys[i / n * p + x], 1)',
+        'cupy_random_multinomial')(xs, p, n, ys)
+    return ys
```
Comment (niboshi) [inline cupy/random/sample.py]: You could support ys.size==0 case by reordering and a condition.

```
    p = len(pvals)
    shape = size + (p,)
    ys = basic.zeros(shape, 'i')
    if ys.size > 0:
        xs = choice(p, p=pvals, size=n * m)
        core.ElementwiseKernel(
            'int64 x, int32 p, int32 n', 'raw int32 ys',
            'atomicAdd(&ys[i / n * p + x], 1)',
            'cupy_random_multinomial')(xs, p, n, ys)
    return ys
```

File: tests/cupy_tests/random_tests/test_sample.py [hunk 2]
```diff
@@ -248,3 +249,33 @@ def test_randn_default_dtype(self):
...
+        x = xp.random.multinomial(10000, pvals, 4)
+        return x / 10000
+
+    @condition.retry(5)
+    @testing.for_float_dtypes()
+    @testing.numpy_cupy_allclose(atol=0.02)
+    def test_size_tuple(self, xp, dtype):
+        pvals = xp.array([0.2, 0.3, 0.5], dtype)
+        x = xp.random.multinomial(10000, pvals, (2, 1))
+        return x / 10000
```
Comment (niboshi) [inline tests/cupy_tests/random_tests/test_sample.py]: Add tests:
- size = (), (0,), (1,0), etc.

**Comment by niboshi** [comment]
Additional comments:
- Write somewhere that `sum(pvals) < 1` case should (and not yet) be supported. For example as `TODO`.
- Returned value is `int32` whereas that of numpy counterpart is `int64`. Shouldn't it be aligned?

**Comment by unnonouno** [comment]
I recently feel this implementation consumes too large memory when `n` is large. But i don't know efficient implementation of multinomial sampler.

- `sum(pvals) < 1`  is todo.
- `atomicAdd` is only supported for `int32`. It is ok to cast it to long.

**Comment by niboshi** [comment]
LGTM.

**Pull request merged**